src/hooks/use-blocker.ts

```diff
@@ -1,16 +1,13 @@
-import { useContext, useEffect } from "react";
+import { useEffect } from "react";
 import { Blocker, Transition } from "history";
-// @ts-ignore
-import { UNSAFE_NavigationContext as NavigationContext } from "react-router-dom";
+import history from "history/browser";
 
 function useBlocker(blocker: Blocker, when = true) {
-  const { navigator } = useContext(NavigationContext);
-
   useEffect(() => {
     if (!when) return;
 
     // @ts-ignore
-    const unblock = navigator.block((tx: Transition) => {
+    const unblock = history.block((tx: Transition) => {
```

Comment:
@maamri95 history instance here would be different than what the application has, isn't it?
We must get the application history, that's the main reason we used navigator from navigation context.
Also, I did checkout on your branch and ran an example application, but it doesn't work for me. Let me know if i am missing something.